pkg/apis/aws/validation/shoot.go

```diff
@@ -52,6 +52,10 @@ func ValidateWorkers(workers []core.Worker, zones []apisaws.Zone, fldPath *field
 			allErrs = append(allErrs, field.Required(fldPath.Index(i).Child("volume"), "must not be nil"))
 		} else {
 			allErrs = append(allErrs, validateVolume(worker.Volume, fldPath.Index(i).Child("volume"))...)
+
+			if worker.Volume.Type != nil && *worker.Volume.Type == "io1" {
```

Comment:
Hmm, I think this will also put out an error, even if the `providerConfig` is set, or am I missing something?

Reply:
Also, I think we should improve the validation logic here, to put out an error, if `.volume.iops` is missing (in case `volumeType=io1` and not only if it is set but not in the valid range: https://github.com/gardener/gardener-extension-provider-aws/blob/f9b9f455c6af7b6e915f3f2de4a438cb4f663bdf/pkg/apis/aws/validation/worker.go#L27-L38

Otherwise the problem can still occur, right?

Reply:
> Hmm, I think this will also put out an error, even if the providerConfig is set, or am I missing something?

Sure, indeed, thank you...

Reply:
The second comment should be already covered by https://github.com/gardener/gardener-extension-provider-aws/blob/master/pkg/validator/shoot_validator.go#L65-L89